src/_pytest/pytester.py

```diff
@@ -650,8 +652,17 @@ def to_text(s):
             p.write(source.strip().encode(encoding), "wb")
             if ret is None:
                 ret = p
+                if ret.strpath not in self._created_files:
```

Comment:
I suggest to replace both lines by:

```python
self._created_files.add(ret)
```

No need to check for membership, and I think keeping a set of `py.path.local` is better. 👍